### PR TITLE
Add mocha (smoke) tests for openhab-cloud

### LINKED ISSUE #113: Smoke tests for openHAB-cloud
In PR #97 and in some others I don't remember anymore, I made some changes and updated the PR with a merge of master and made a simple small mistake during the conflict resolution. While I would agree, that I would need to be more careful about these little kind of things (:P), I would love some smoke tests more :D

In my vision the tests would try to run app.js and look at any error message during startup and fail, if there're any and stop the instance and proceed with unit tests after it. We could probably extend the tests with requesting some endpoints later, however, just starting app.js sounds like a good idea for now.

### PR DESCRIPTION
**Pull request by FlorianSW**
The mocha tests can be used to test a locally started openhab-cloud instance (request/response based), which is an addition to unit tests (which tests single functions in isolation). The current set of mocha tests are really simple and just test the main page and a login with a correct and incorrect password.

Fixes #113

### CODE REVIEW

File: tests/mocha/login.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+var supertest = require('supertest');
+var should = require('should');
+var lib = require('../lib');
+var user = require('../../models/user');
+
+var ohCloud = supertest.agent('http://localhost:3000');
+
+describe('Test login', function() {
+    before('create necessary user', function (done) {
+        user.register('test', '123', done);
```
Comment (FlorianSW) [inline tests/mocha/login.js]: For whatever reason, done() is not called until the timeout of 2 seconds is reached, but also a timeout of 15 seconds does not work, so I assume, that there's another error which prevents the callback for user.register to be called. However, I was not able to figure out why. If someone has an idea, I'm open for suggestions :)

**Comment by marziman** [comment]
@FlorianSW I have proposed mocha long time before for testing and that was sth. I wanted to tackle. So we are not that far away from each other :-)
I will take alook as soon as possible..

**Comment by FlorianSW** [comment]
I was looking at different possible approaches (the simplest one was a list of urls and the expected status code), however, mocha is, from my small research and point of view, the best option, as it gives us flexibility  about what we want to test, and also allows us to have simple, small smoke tests (like the main page test) :)

I'm happy to read your comments! :)

**Comment by marziman** [comment]
@FlorianSW absolutely cool. So I already have evaluated it before and suggested it to Dan, but we went into other trouble back than. So thanks for going on with that..
I will set it up and try it out. Next, since you added the initial part, I will do some more test cases.

Only one small remark: Maybe we need to write later on, if all runs well, some small intro how to extend the tests/testcases for the community.

**Comment by digitaldan** [comment]
👍  for mocha tests!

**Comment by marziman** [comment]
Merging, Fixes https://github.com/openhab/openhab-cloud/issues/113

**Pull request merged**